src/components/app.jsx

```diff
@@ -1,43 +1,106 @@
 import React from 'react';
+import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
 import Header from './header/header';
 import contactsData from '../data';
 import ToolBar from './toolBar/toolBar';
 import ContactList from './contactList/contactList';
+import EditForm from './editForm/editForm';
 
 
 export default class App extends React.Component {
   state = {
     contacts: contactsData,
+    filteredContacts: '',
+    editableContact: '',
   };
 
   handleInputChange = (e) => {
     const { value } = e.target;
+    const { contacts } = this.state;
 
-    const filteredContacts = contactsData.filter((contact) => {
+    const filteredContacts = contacts.filter((contact) => {
       const name = contact.name.toLowerCase().replace(/ /g, '');
       return name.includes(value.toLowerCase());
     });
 
     this.setState({
-      contacts: filteredContacts,
+      filteredContacts,
+    });
+  };
+
+  addNewContact = ({ name, phone }) => {
+    const { contacts } = this.state;
+    const newID = contacts.length ? (contacts[contacts.length - 1].id + 1) : 0;
+    const newContact = {
+      id: newID,
+      name,
+      phone,
+    };
+
+    this.setState({
+      contacts: [...contacts, newContact],
+      filteredContacts: '',
     });
   };
 
   deleteContact = (id) => {
-    this.setState(prevState => (
-      { contacts: prevState.contacts.filter(contact => contact.id !== id) }));
+    const { contacts } = this.state;
+
+    this.setState({
+      contacts: contacts.filter(contact => contact.id !== id),
+      filteredContacts: '',
+    });
   };
 
+  findContact = (id) => {
+    const { contacts } = this.state;
 
-  render() {
+    this.setState({
+      editableContact: contacts.find(contact => contact.id === id),
```

Comment:
I don't think we need to store it in state

Reply:
Fixed